### PR TITLE
Fix request count check for BadHttpMethod

### PR DESCRIPTION
**Pull request by bdraco**
The error handler was incorrectly checking for the first request of the server instead of the first request for the connection

https://github.com/aio-libs/aiohttp/pull/10055#discussion_r1864316780

### CODE REVIEW

File: aiohttp/web_protocol.py [hunk 1]
```diff
@@ -212,6 +212,9 @@ def __init__(
     ):
         super().__init__(loop)
 
+        # _request_count is the number of request processed by the handler
+        # A new handler is created for each connection, so it is the number
+        # of requests processed by the connection.
```
Comment (Dreamsorcerer) [inline aiohttp/web_protocol.py]: I find this slightly confusing. I'd probably avoid the word handler here as I'm thinking of the endpoint handler (i.e. the function registered on a route) which is presumably not what you mean.

Comment (bdraco) [inline aiohttp/web_protocol.py]:
```suggestion
        # _request_count is the number of requests processed with the same connection.
```

Comment (bdraco) [reply aiohttp/web_protocol.py]: I simplified the comment. Hopefully it makes more sense now

**Review by Dreamsorcerer — COMMENTED** [review]
I'm still not convinced this will behave as desired. Clients that don't use keep alive (whether unsupported, or a proxy doesn't allow it etc.) will never result in the issue getting logged.
While anyone using a reverse proxy like nginx will likely gets loads of logs as the "client" is the proxy server, not the connecting user.

**Comment by bdraco** [comment]
> I'm still not convinced this will behave as desired. Clients that don't use keep alive (whether unsupported, or a proxy doesn't allow it etc.) will never result in the issue getting logged. While anyone using a reverse proxy like nginx will likely gets loads of logs as the "client" is the proxy server, not the connecting user.

If I remember correctly `nginx` has keep-alive disabled by default for reverse proxying..maybe that has changed https://nginx.org/en/docs/http/ngx_http_upstream_module.html#keepalive

I'm not sure what you are suggesting for the path forward.

**Comment by Dreamsorcerer** [comment]
> > I'm still not convinced this will behave as desired. Clients that don't use keep alive (whether unsupported, or a proxy doesn't allow it etc.) will never result in the issue getting logged. While anyone using a reverse proxy like nginx will likely gets loads of logs as the "client" is the proxy server, not the connecting user.
> 
> If I remember correctly `nginx` has keep-alive disabled by default for reverse proxying..maybe that has changed https://nginx.org/en/docs/http/ngx_http_upstream_module.html#keepalive

I think it's still very common to enable keepalive. Regardless, when disabled we'll never log, when enabled we'll almost always log. Neither is the intended behaviour.

> I'm not sure what you are suggesting for the path forward.

Other than logging unconditionally as before, I don't have a better suggestion. Maybe when that other issue is fixed, users could customise this with a middleware?

**Comment by Dreamsorcerer** [comment]
With a middleware they could choose to opt in to this behaviour, they could use the client IP from a trusted proxy header instead of per-connection etc.

**Comment by bdraco** [comment]
> Other than logging unconditionally as before, I don't have a better suggestion. Maybe when that other issue is fixed, users could customise this with a middleware?

Yeah I think customization in https://github.com/aio-libs/aiohttp/issues/3287 probably the best way to handle this in the longer term. 

I think this PR is ok for the short term as its still logged at debug level so it should be discoverable if the user increases the logging level but considering this generates 100+ issues downstream when nothing is actually wrong I'd prefer to keep it at debug level as a log message isn't so valuable when the common case is noise.

**Comment by Dreamsorcerer** [comment]
Yep, I think probably merge this for now as something reasonable.

I think #3287 will be a breaking change and won't get backported (which is why I put in on the v4 milestone). So, then we can drop the debug logs in v4 when that is ready.

**Pull request merged**